src/strategies/FluentStrategy.ts

```diff
@@ -0,0 +1,104 @@
+import {Arbitrary, FluentPick} from '../arbitraries'
+import {FluentConfig, FluentResult} from '../FluentCheck'
+import {StrategyArbitraries} from './types'
+
+export interface FluentStrategyInterface {
+  hasInput: <K extends string>(arbitraryName: K) => boolean
+  getInput: <K extends string, A>(arbitraryName: K) => FluentPick<A>
+  handleResult: () => void
+}
```

Comment:
Why do you need the K generic if it's a string? Is there something that actually extends string?

Reply:
Also, `Iterator` in typescript is defined as:

```typescript
interface Iterator<T> {
    next(value?: any): IteratorResult<T>;
    return?(value?: any): IteratorResult<T>;
    throw?(e?: any): IteratorResult<T>;
}

interface IteratorResult<T> {
    done: boolean;
    value: T;
}
```

Does it make sense for `FluentStrategyInterface` to extend an `Iterator` of `FluentPick`'s? Food for thought.

Reply:
The use of K was to maintain coherence with the code already developed by you guys, which also uses `K extends String`

Reply:
> The use of K was to maintain coherence with the code already developed by you guys, which also uses `K extends String`

We do that because we want to use 'string''s as singleton types in 'record''s. I'm not sure if it is the case here, but I'll check it out later, that's ok.

Reply:
Ok, feel free to check that out and let me know what is your final suggestion?